### ISSUE TITLE
Installing on Linux via WINE and then running fails

### ISSUE THREAD

**Issue by ThornedRose** [Xjph/ObservatoryCore#102]
I use Arch Linux with KDE. I installed .Net 6.0 into my default WINE prefix and installed ObservatoryCore fine. However, running the exe (from the desktop icon) shows a KDE loading animation and then that disappears (fails to load). Are there any dependencies beyond .Net 6 that I are required to run this? 

If I try to run from a terminal, here's what I get:

`$ wine ObservatoryCore.exe 
0580:fixme:heap:GetNumaHighestNodeNumber semi-stub: 000000000019EFDC
0580:fixme:seh:WerRegisterRuntimeExceptionModule (L"C:\\Program Files\\dotnet\\shared\\Microsoft.NETCore.App\\6.0.12\\mscordaccore.dll", 0000000001290000) stub
0580:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0580:fixme:heap:RtlSetHeapInformation handle 00000000240B0000, info_class 0, info 0000000000198CB0, size 4 stub!
0580:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 00000000249225B7, 28) stub
0594:fixme:ver:GetCurrentPackageId (0000000023EDFE10 0000000000000000): stub
0580:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F20F10, 34) stub
0580:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F187C8, 17) stub
0580:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F1DBB8, 38) stub
0580:fixme:ntdll:NtQuerySystemInformation (0x000000ce,0x19cdd0,0x00000008,(nil)) stub
0580:fixme:nls:get_dummy_preferred_ui_language (0x8 0x1409 000000000019DD10 0000000000000000 000000000019DD08) returning a dummy value (current locale)
0580:fixme:nls:get_dummy_preferred_ui_language (0x8 0x1409 000000000019DD10 000000000019DC60 000000000019DD08) returning a dummy value (current locale)
0580:fixme:ntdll:RtlGetCurrentProcessorNumberEx (00000000001982A0) :semi-stub
`
<snip - this line is repeated 160 more times>
`
0580:fixme:ntdll:RtlGetCurrentProcessorNumberEx (0000000000199660) :semi-stub
0580:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F19458, 51) stub
0580:err:virtual:virtual_setup_exception stack overflow 2000 bytes addr 0x170031350 stack 0x20830 (0x20000-0x21000-0x1a0000)
^C05dc:fixme:console:default_ctrl_handler Terminating process 55c on event 0
056c:fixme:ver:GetCurrentPackageId (000000000021FDB0 0000000000000000): stub
`
Thanks!

**Comment by ThornedRose** [comment]
I've tried both. Above is from the standalone. The installer works fine, but when trying to run it just opens a box for a moment which I then can't see and the icon sits in my task bar but I'm unable to interact with it.

FWIW, here's the terminal output from the installed version:

```
wine ObservatoryCore.exe 
0048:fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {00000000-0000-0000-0000-000000000000}
0048:fixme:process:SetProcessShutdownParameters (00000280, 00000001): partial stub.
0048:fixme:heap:RtlSetHeapInformation handle 00000000, info_class 1, info 00000000, size 0 stub!
0048:fixme:userenv:GetProfileType 0053FDE8
0060:fixme:advapi:RegisterEventSourceW ((null),L"edgeupdate"): stub
0060:fixme:advapi:ReportEventW (CAFE4242,0x0004,0x0000,0x00000000,00000000,0x0001,0x00000000,0142FC90,00000000): stub
0060:fixme:advapi:DeregisterEventSource (CAFE4242) stub
0048:fixme:kernelbase:AppPolicyGetProcessTerminationMethod FFFFFFFA, 0053FEAC
0070:fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {00000000-0000-0000-0000-000000000000}
0070:fixme:process:SetProcessShutdownParameters (00000280, 00000001): partial stub.
0070:fixme:heap:RtlSetHeapInformation handle 00000000, info_class 1, info 00000000, size 0 stub!
0070:fixme:userenv:GetProfileType 0053FDE8
0070:fixme:process:WTSGetActiveConsoleSessionId stub
0070:fixme:wtsapi:WTSQuerySessionInformationW Unimplemented class 8
0070:fixme:wtsapi:WTSEnumerateSessionsW Stub 00000000 0x00000000 0x00000001 0053FBA4 0053FBA8
00b8:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
00b8:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
00b8:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
00b8:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
00b8:fixme:wineusb:query_id Unhandled ID query type 0x5.
00b8:fixme:wineusb:query_id Unhandled ID query type 0x5.
00b8:fixme:wineusb:query_id Unhandled ID query type 0x5.
00b8:fixme:wineusb:query_id Unhandled ID query type 0x5.
00b8:fixme:wineusb:query_id Unhandled ID query type 0x5.
00b8:fixme:wineusb:query_id Unhandled ID query type 0x5.
00b8:fixme:wineusb:query_id Unhandled ID query type 0x5.
00b8:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
00b8:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
002c:fixme:ver:GetCurrentPackageId (000000000021FDB0 0000000000000000): stub
0034:fixme:ver:GetCurrentPackageId (000000000021FDB0 0000000000000000): stub
0150:fixme:heap:GetNumaHighestNodeNumber semi-stub: 000000000019EFDC
0150:fixme:seh:WerRegisterRuntimeExceptionModule (L"C:\\Program Files\\dotnet\\shared\\Microsoft.NETCore.App\\6.0.12\\mscordaccore.dll", 0000000001290000) stub
0150:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0164:fixme:ver:GetCurrentPackageId (0000000023EDFE10 0000000000000000): stub
0150:fixme:heap:RtlSetHeapInformation handle 00000000240B0000, info_class 0, info 0000000000198CB0, size 4 stub!
0150:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 00000000249225B7, 28) stub
0150:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F20F10, 34) stub
0150:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F187C8, 17) stub
0150:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F1DBB8, 38) stub
0150:fixme:ntdll:NtQuerySystemInformation (0x000000ce,0x19cdd0,0x00000008,(nil)) stub
0150:fixme:nls:get_dummy_preferred_ui_language (0x8 0x1409 000000000019DD10 0000000000000000 000000000019DD08) returning a dummy value (current locale)
0150:fixme:nls:get_dummy_preferred_ui_language (0x8 0x1409 000000000019DD10 000000000019DC60 000000000019DD08) returning a dummy value (current locale)
0150:fixme:ntdll:RtlGetCurrentProcessorNumberEx (00000000001982A0) :semi-stub
```
<snip. line is repeated 100+ more times>
```
0150:fixme:ntdll:RtlGetCurrentProcessorNumberEx (0000000000199660) :semi-stub
0150:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000022F19458, 51) stub
0150:err:virtual:virtual_setup_exception stack overflow 2000 bytes addr 0x170031350 stack 0x20830 (0x20000-0x21000-0x1a0000)
0194:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0194:fixme:seh:WerSetFlags (2) stub
0194:fixme:heap:RtlSetHeapInformation handle 00000000, info_class 1, info 00000000, size 0 stub!
01b0:fixme:heap:RtlSetHeapInformation handle 0000000001080000, info_class 0, info 000000000011F680, size 4 stub!
01b0:fixme:seh:WerSetFlags (2) stub
01b0:fixme:heap:RtlSetHeapInformation handle 0000000000000000, info_class 1, info 0000000000000000, size 0 stub!
```

The original Elite Observatory works fine (apart from TTS which I am still working on getting working in Wine). So I guess the different .NET version is screwing something up. I don't have much experience with running .NET 5+ apps in Wine though - all the apps I've used thus far are .NET 4.x and work fine for the most part.

**Comment by Xjph** [comment]
Given that both of your outputs have a _lot_ of `fixme` annotations it certainly seems like this is probably something the various Wine developers are aware of. Unfortunately I don't really have any suggestions for how to resolve this in the here and now.

That said, it is _ostensibly possible_ to build a linux native version, .NET 5+ supports linux as a build target. I've had some minor success building and running earlier versions of Observatory Core on Ubuntu, but there were some UI problems that while they didn't make it completely unusable were severe enough to prevent me from officially providing a release. Depending on your comfort level with building from source you might be able to try that, or I might be able to provide a test release for you when I have some time to dig into it myself.

**Comment by cerus** [comment]
I've just tried building this repo and it's working great so far on Arch Linux. Haven't done a lot of testing yet though. There's only one small issue that I've noticed: The popup notifications take the cursor focus away from Elite. Aside from that everything works as expected.
I've created [a little repo](https://github.com/cerus/observatorycore-linux) that contains prebuilt executables and a build script, maybe it can be of use for others :) (@Xjph let me know if you're not comfortable with that and I'll take it down)
